### PR TITLE
Assign parking stalls to households

### PR DESCRIPTION
**Pull request by haitamlaarabi**
The reservedFor column in the parking file now serves as a way to assign stalls to households, by using the format: type(managedId), we can make difference between household(household-id-0) and ridehail(a-ridehail-company) or shared(shared-fleet-company).

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3294)

### CODE REVIEW

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

File: src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala [hunk 1]
```diff
@@ -1,31 +1,95 @@
...
+    var reservedForMaybe = reservedForString match {
+      case null | "" =>
+        Some(VehicleManager.AnyManager)
+      case x if x == VehicleManager.AnyManager.managerId.toString =>
+        Some(createOrGetReservedFor(reservedForString, TypeEnum.Default))
+      case CustomReservedForRegex(kind, id) =>
+        Some(createOrGetReservedFor(id, TypeEnum.withName(kind.trim.toLowerCase)))
+      case _ =>
+        None
+    }
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala:67]: These 10 lines can be replaced with the call of `initReservedFor`

File: src/main/scala/beam/agentsim/infrastructure/parking/ParkingZoneFileUtils.scala [hunk 2]
```diff
@@ -488,37 +466,33 @@ object ParkingZoneFileUtils extends ExponentialLazyLogging {
...
+  private def calculateNumStalls(
+    initialNumStalls: Double
+  )(implicit parkingStallCountScalingFactor: Double, rand: Random): Int = {
+    val expectedNumberOfStalls = initialNumStalls * parkingStallCountScalingFactor
+    val floorNumberOfStalls = math.floor(expectedNumberOfStalls).toInt
+    if ((expectedNumberOfStalls % 1.0) > rand.nextDouble) {
+      floorNumberOfStalls + 1
+    } else {
+      floorNumberOfStalls
+    }
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/infrastructure/parking/ParkingZoneFileUtils.scala]: We could use beam.utils.MathUtils#roundUniformly for the same purpose. Though in this case a new overloaded version of this method is required: the one that gets an instance of Random as an input parameter.

**Review by dimaopen — COMMENTED** [review]
2 minor notices

**Review by nikolayilyin — COMMENTED** [review]
> Reviewed 52 of 62 files at r1, 4 of 4 files at r2, 7 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3294)* status: all files reviewed, 3 unresolved discussions (waiting on @haitamlaarabi and @JustinPihony)

___
*[src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala, line 71 at r3](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3294#-Miq2uLH7qBRgulE0vT0:-Miq2uLI4K8YD7LPkRNi:bh6ojfs) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/7a300893abf3272742f37fb32377a2e48e60a460/src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala#L71)):*
> ```Scala
> 
>   def createOrGetReservedFor(reservedForString: String, beamConfigMaybe: Option[BeamConfig]): Option[ReservedFor] = {
>     var reservedForMaybe = reservedForString match {
> ```

Can we use here the initReservedFor function call instead of pattern matching? 
The pattern matching looks identical to the code of the initReservedFor function

**Review by haitamlaarabi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3294)* status: all files reviewed, 3 unresolved discussions (waiting on @dimaopen, @JustinPihony, and @nikolayilyin)

___
*[src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala, line 71 at r3](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3294#-Miq2uLH7qBRgulE0vT0:-Mj1qA6CBiEaJ1gT0mf3:b-wsc9c3) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/7a300893abf3272742f37fb32377a2e48e60a460/src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala#L71)):*
> initReservedFor
I have just realized that I'm not using initReservedFor. So I just deleted that function. Thanks
___
*[src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala, line 80 at r3](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3294#-Miei-f6olmU0u7jypFr-r3-80:-Mj1qNwP0wc_X16A3tqq:bim3daq) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/7a300893abf3272742f37fb32377a2e48e60a460/src/main/scala/beam/agentsim/agents/vehicles/VehicleManager.scala#L80)):*
> I have just realized that I'm not using initReservedFor. So I just deleted that function. Thanks
Yes, I have just realized that I'm not using initReservedFor. So I just deleted that function. Thanks
___
*[src/main/scala/beam/agentsim/infrastructure/parking/ParkingZoneFileUtils.scala, line 478 at r3](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3294#-Miei-f7xwIzth2vpU2y-r3-478:-Mj1s80t1VMoVLXaTRky:b-llqeg9) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/7a300893abf3272742f37fb32377a2e48e60a460/src/main/scala/beam/agentsim/infrastructure/parking/ParkingZoneFileUtils.scala#L478)):*
<details><summary><i>Previously, dimaopen (Dmitrii Openkov) wrote…</i></summary><blockquote>

We could use beam.utils.MathUtils#roundUniformly for the same purpose. Though in this case a new overloaded version of this method is required: the one that gets an instance of Random as an input parameter.
</blockquote></details>

That's true. Done!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Review by dimaopen — APPROVED** [review]

**Pull request merged**